Tests/L0/SonarQubePostTest/PRCA/PostCommentsTests.ps1

```diff
@@ -284,37 +361,96 @@ function InitPostCommentsModule
 }
 
 #
-# Test - Post messages where existing messages are already present
+# Test - E2E test that goes through several iterations of posting messages. Note  
 #
 
 # Arrange
 $mockDiscussionClient = InitPostCommentsModule
-$modifiedFilesinPr = @("some/path1/file.cs", "some/path2/file.cs")
-
+$modifiedFilesinPr = @("some/path1/file.cs", "some/path2/file.cs") # p1 and p2 are the only files modified by this PR
 Register-Mock GetModifiedFilesInPR { $modifiedFilesinPr }
 
-$comment1 = BuildTestMessage "CA issue 1" 14 "some/path1/file.cs" 1 
-$comment2 = BuildTestMessage "CA issue 2" 15 "some/path2/file.cs" 5
-$comment3 = BuildTestMessage "CA issue 3" 15 "some/path3/file.cs" 5  # issue in a file not changed by the PR 
+# Iteration 1: 
+# - Current state: {no existing comments}
+# - Messages to be posted: path1:A, path2:B, otherPath:C
+# - Expected comments after posting: path1:A, path2:B
+
+$p1A = BuildTestMessage "A" 14 "some/path1/file.cs" 1 
```

Comment:
I re-wrote the test to make it easier to read. It was of great help for ironing out bugs